wire-ios/Wire-iOS/Sources/UserInterface/DeviceView/Views/E2EIdentityCertificateDetailsView.swift

```diff
@@ -0,0 +1,149 @@
+            label: {
+                Image(.download)
+                    .foregroundColor(SemanticColors.Icon.foregroundDefaultBlack.swiftUIColor)
+            }
+        )
+    }
+
+    private var downloadButton: some View {
+        SwiftUI.Button(
+            action: {
+                performDownload?()
+            },
+            label: {
+                Text(L10n.Localizable.Content.Message.download)
+                    .font(FontSpec.normalBoldFont.swiftUIFont)
+            }
+        )
+    }
+
+    private var moreButton: some View {
+        SwiftUI.Button(
+            action: {
+                isMenuPresented.toggle()
+            },
+            label: {
+                Image(.more)
+                    .foregroundColor(SemanticColors.Icon.foregroundDefaultBlack.swiftUIColor)
+                    .padding(.trailing, ViewConstants.Padding.standard)
+            }
+        )
+    }
+
+    private var copyToClipboardButton: some View {
+        SwiftUI.Button(
+            action: {
+                performCopy?(certificateDetails)
+            },
+            label: {
+                Text(L10n.Localizable.Device.Details.CertificateDetails.copyToClipboard)
+                    .foregroundColor(SemanticColors.Label.textDefault.swiftUIColor)
+            }
+        )
+    }
+
+    private var bottomBarView: some View {
+        VStack {
+            if isDownloadAndCopyEnabled {
+                HStack {
+                    downloadImageButton.padding()
+                    downloadButton
+                        .foregroundColor(SemanticColors.Icon.foregroundDefaultBlack.swiftUIColor)
+                        .padding()
+                    Spacer()
+                    moreButton
+                        .foregroundColor(SemanticColors.Icon.foregroundDefaultBlack.swiftUIColor)
+                    .confirmationDialog("...", isPresented: $isMenuPresented) {
+                        copyToClipboardButton
+                            .foregroundColor(SemanticColors.Icon.foregroundDefaultBlack.swiftUIColor)
+                    }
+                }
+            }
+        }
+        .padding()
+        .frame(maxWidth: .infinity)
+        // The background will extend automatically to the edge
+        .overlay(alignment: .top) {
+            if isDownloadAndCopyEnabled {
+                Rectangle()
+                    .frame(height: 1)
+                    .foregroundColor(SemanticColors.View.backgroundSeparatorCell.swiftUIColor)
+            }
+        }
+    }
+
+    var body: some View {
+        titleView.background(SemanticColors.View.backgroundDefault.swiftUIColor)
+        certificateView.background(SemanticColors.View.backgroundDefaultWhite.swiftUIColor)
+        .safeAreaInset(edge: .bottom, spacing: .zero) {
+            bottomBarView.background(SemanticColors.View.backgroundUserCell.swiftUIColor)
+        }
+        .ignoresSafeArea()
+        .background(SemanticColors.View.backgroundDefault.swiftUIColor)
+    }
```

Comment:
nitpic: return to new line chain actions for SwiftUI

Reply:
@theswiftway could you apply this suggestion?

Reply:
I will do this part of [WPB-3324](https://wearezeta.atlassian.net/browse/WPB-3324)

Reply:
done